### PR TITLE
Update of docs and test data

### PR DESCRIPTION
**Pull request by wdwvt1**
filter_fasta.py contained ambiguous documentation for the --sample_id_fp option. Specifically, it was unclear what format the input file for the --sample_id_fp option should be in. This commit does three minor things:
1. Updates the help text for this option so its clear what format is expected.
2. Updates the qiime_test_data/filter_fasta/map.txt so that it conforms to the new documentation and actually exercises the functionality. The current master branch code filters out all sequences.
3. Makes the help options in the script documentation stylistically similar, i.e. same capitalization, punctuation, etc.

### CODE REVIEW

**Comment by qiime-r2d2** [comment]
Build results will soon be (or already are) available at: http://ci.qiime.org/job/qiime-github-pr/1596/

File: scripts/filter_fasta.py [hunk 1]
```diff
@@ -63,25 +63,31 @@
...
 script_info['optional_options'] = [
     make_option('-m', '--otu_map', type='existing_filepath',
-                help='an OTU map where sequences ids are those which should be retained'),
+                help="An OTU map where sequences ids are those which should be "
+                     "retained."),
     make_option('-s', '--seq_id_fp', type='existing_filepath',
-                help='A list of sequence identifiers (or tab-delimited lines with'
-                ' a seq identifier in the first field) which should be retained'),
+                help="A list of sequence identifiers (or tab-delimited lines "
+                     "with a seq identifier in the first field) which should "
```
Comment (ElDeveloper) [inline scripts/filter_fasta.py]: there's a missing `be` in this line.

**Comment by qiime-r2d2** [comment]
Build results will soon be (or already are) available at: http://ci.qiime.org/job/qiime-github-pr/1598/

**Comment by ElDeveloper** [comment]
@wdwvt1, this looks good, though I just noticed that map.txt is used by two usage examples. One of them expecting an OTU map and the other expecting the `--sample_id_fp` format. I think the best thing here would be to have separate files.

That being said map.txt looked a lot like a mapping file not like any of the two files it is supposed to represent.

**Comment by wdwvt1** [comment]
@ElDeveloper - I think `map.txt` is only used in one usage example. `otu_map.txt` is used in another usage example (the first one).

**Comment by ElDeveloper** [comment]
Your are exactly right, sorry that's totally my bad! Thanks for putting this fix in!

**Pull request merged**